From 884f7b3e917194ebb3d7e621df9af7ed496a91eb Mon Sep 17 00:00:00 2001
From: Olivier Masse <olivier.masse@nxp.com>
Date: Wed, 16 Nov 2022 12:05:50 +0100
Subject: [PATCH 1/3] LFOPTEE-126: Add spl and sld Blocks info

Dump hab block information used by the signature script.
To ease the parsing process in meta-secure-boot recipe,
mkimage_imx8 tool dump spl and sld hab blocks correctly
formated for csf configuration file.

Signed-off-by: Olivier Masse <olivier.masse@nxp.com>

Upstream-Status: Backport [66cef04afacc104e47fb65ac9879e70e45334c3f]
Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 iMX8M/mkimage_imx8.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/iMX8M/mkimage_imx8.c b/iMX8M/mkimage_imx8.c
index 54828d1..06ab485 100644
--- a/iMX8M/mkimage_imx8.c
+++ b/iMX8M/mkimage_imx8.c
@@ -1662,6 +1662,14 @@ int main(int argc, char **argv)
 	fprintf(stderr, " sld hab block: \t0x%x 0x%x 0x%x\n",
 		sld_load_addr, sld_header_off, sld_csf_off - sld_header_off);
 
+	fprintf(stderr, "SPL CSF block:\n");
+	fprintf(stderr, "\tBlocks = \t0x%x 0x%x 0x%x \"flash.bin\"\n",
+		imx_header[IMAGE_IVT_ID].fhdr.self, header_image_off, csf_off - header_image_off);
+
+	fprintf(stderr, "SLD CSF block:\n");
+	fprintf(stderr, "\tBlocks = \t0x%x 0x%x 0x%x \"flash.bin\",\\\n",
+		sld_load_addr, sld_header_off, sld_csf_off - sld_header_off);
+
 	return 0;
 }
 
-- 
2.42.0

